server/controllers/kaban.js

```diff
@@ -0,0 +1,287 @@
+  }
+  if (board.userId != userId) {
+   res.status(401)
+   return res.send("Hey!😁, you are not authorized")
+  }
+  let column = await updateBoardColumn({ name, index }, board_id, column_id);
+  res.status(200)
+  res.json({ success: true, column })
+ } catch (error) {
+  res.status(501)
+  throw new Error("Error Processing your request at this time")
+ }
+});
+
+// @route PUT /column/:board_id
+// @access Protected
+exports.deleteColumn = asyncHandler(async (req, res,) => {
+ try {
+  const userId = req.user.id;
+  const { board_id, column_id } = req.params;
+  let board = await Board.findOne({ _id: board_id });
+  if (!board) {
+   res.status(401)
+   return res.send("Column's Board does not exist")
+  }
+  if (board.userId != userId) {
+   res.status(403)
+   return res.send("Hey!😁, you are not authorized")
+  }
+  await deleteBoardColumn(board_id, column_id);
+  res.status(200)
+  res.json({ success: true })
+ } catch (error) {
+  res.status(501)
+  throw new Error("Error Processing your request at this time")
+ }
+});
+
+// @route POST /kaban/task/:board_id/:column_id
+// @access Protected
+exports.createTask = asyncHandler(async (req, res,) => {
+ try {
+  const userId = req.user.id;
+  const { name, description, index } = req.body;
+  const { board_id, column_id } = req.params;
+
+  if (index && typeof index != 'number') {
+   res.status(401)
+   return res.send("Index value not supported")
+  }
+
+  let board = await Board.findOne({ _id: board_id });
+  if (!board) {
+   res.status(403)
+   return res.send("Column's Board does not exist")
+  }
+  if (board.userId != userId) {
+   res.status(403)
+   return res.send("Hey!😁, you are not authorized")
+  }
+  let column = await Column.findOne({ _id: column_id });
+  if (!column) {
+   res.status(403)
+   return res.send("Task's Column does not exist")
+  }
+  let task = await createColumnTask({ name, description, index }, column_id);
+  console.log("task", task);
+  res.status(200)
+  res.json({ success: true, task })
+ } catch (error) {
+
+  res.status(501)
+  throw new Error("Error Processing your request at this time")
+ }
+});
+
+// @route PUT /kaban/task/:board_id/:column_id/:task_id
+// @access Protected
+exports.updateTask = asyncHandler(async (req, res,) => {
+ try {
+  const userId = req.user.id;
+  const { name, description, targetColumnId, index } = req.body;
+  const { board_id, column_id: current_column_id, task_id } = req.params;
+
+  if (targetColumnId && !(index >= 0)) {
+   res.status(401)
+   return res.send("Index must be specified in order to use a diffent ColumnId")
+  }
+  if (index && typeof index != 'number') {
+   res.status(401)
+   return res.send("Index value not supported")
+  }
+
+  let board = await Board.findOne({ _id: board_id });
+  if (!board) {
+   res.status(403)
+   return res.send("Column's Board does not exist")
+  }
+  if (board.userId != userId) {
+   res.status(403)
+   return res.send("Hey!😁, you are not authorized")
```

Comment:
This is a `401`. Chain these together. No need for a `return` statement as `res.send` already terminates request/response cycle.